src/schedule/zephyr_ll.c

```diff
@@ -198,19 +199,25 @@ static void zephyr_ll_run(void *data)
 			continue;
 		}
 
-		pdata->run = true;
-		task->state = SOF_TASK_STATE_RUNNING;
+		if (domain_is_pending(domain, task, NULL)) {
+			pdata->run = true;
+			task->state = SOF_TASK_STATE_RUNNING;
+		}
 
 		/* Move the task to a temporary list */
 		list_item_del(list);
 		list_item_append(list, &task_head);
 
+		if (task->state != SOF_TASK_STATE_RUNNING)
```

Comment:
ditto, re inline comments - scheduler can be complex so lets add comments to help.

Reply:
Done.

Reply:
Also, I can see the fails in CI, and it's related intel test. But I can't do the test in my side cause we have no intel board.
Can you please help? I also will look the patch to see if any I can change to fix the CI fails.

Reply:
Ack, @TangleZ I'll try to debug this a bit early next week, agreed this is difficult to debug without a board available.

Reply:
Thanks for your help!